### PR TITLE
Time Profiler utility module.

### PR DESCRIPTION
**Pull request by gmagannaDevelop**
I've added a file to utils/. This file "timing.py" contains wrappers useful for function execution profiling.

### CODE REVIEW

File: segnet/utils/timing.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+
+import os
+import json
+import datetime
+
+from functools import wraps
+from time import time
+from typing import Callable
+
+def time_this(f) -> Callable:
```
Comment (edwinb-ai) [inline segnet/utils/timing.py]: It seems like you annotate most of the code, which is fine, so this should also be annotated?

File: segnet/utils/timing.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            with open(path_to_logfile, 'a') as log:
+                log.write(json.dumps(data)+'\n')
+            return result
+        ##
+        return wrap
+    ##
+    return timed
+##
+
+def is_jsonable(x):
```
Comment (edwinb-ai) [inline segnet/utils/timing.py]: I think the same applies here, no annotations, should put them in place just to be consistent.

File: segnet/utils/timing.py (hunk 1 above)
Comment (gmagannaDevelop) [reply segnet/utils/timing.py]: Right on! I added the annotations at the beginnig, then forgot to annotate the rest of the code.

File: segnet/utils/timing.py (hunk 2 above)
Comment (gmagannaDevelop) [reply segnet/utils/timing.py]: This should also be annotated.

**Comment by edwinb-ai** [comment]
Look good to me! When Travis CI finishes, I will merge this. Thanks a lot! :tada: :octocat:

**Pull request merged**